app/src/common/shared/com/igalia/wolvic/utils/EnvironmentsManager.java

```diff
@@ -116,19 +116,29 @@ private void downloadEnvironment(@NonNull String envId) {
         final Environment environment = EnvironmentUtils.getExternalEnvironmentById(mContext, envId);
         final String payload = EnvironmentUtils.getEnvironmentPayload(environment);
         if (environment != null) {
-            // Check if the env is being downloaded
-            boolean isDownloading = mDownloadManager.getDownloads().stream()
-                    .filter(item ->
-                            item.getStatus() == DownloadManager.STATUS_RUNNING &&
-                                    item.getStatus() == DownloadManager.STATUS_PAUSED &&
-                                    item.getStatus() == DownloadManager.STATUS_PENDING &&
-                                    item.getUri().equals(payload))
-                    .findFirst().orElse(null) != null;
-
-            if (!isDownloading) {
-                // If the env is not being downloaded, start downloading it
+            // Check if the env is being downloaded or has been already downloaded.
+            // The download item will be removed in 2 situations:
+            //   1- the user selects another env before the download is completed (see setOrDownloadEnvironment)
+            //   2- when the unzip task is completed successfully.
+            Download envItem = mDownloadManager.getDownloads().stream()
+                    .filter(item -> item.getUri().equals(payload))
+                    .findFirst().orElse(null);
+
+            if (envItem == null) {
```

Comment:
We don't use envItem at all so I guess we can keep the previous code?

Reply:
We use it to get the Id in the ```else if (envItem.getStatus() == Download.SUCCESSFUL) ``` branch.